sdk/core/Azure.Core/src/Pipeline/HttpPipelineSynchronousPolicy.cs

```diff
@@ -20,11 +38,23 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
         }
 
         /// <inheritdoc />
-        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
         {
-            OnSendingRequest(message);
-            await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
-            OnReceivedResponse(message);
+            async ValueTask ProcessAsyncInner(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+            {
+                OnSendingRequest(message);
```

Comment:
Is this creating a separate capture class, though, which would cause allocs? If not (just declaring a method at compile-time on this class), fine.

Reply:
There is no additional capture here outside of the async state machine.